tests/unit/customizations/test_preview.py

```diff
@@ -40,14 +40,33 @@ def test_invoke_preview_mode_service(self):
         # ever mark cloudfront as not being a preview service
         # by default.
         self.assertIn('cloudfront', preview.PREVIEW_SERVICES)
-        rc = self.driver.main('cloudfront help'.split())
+        rc = self.driver.main('cloudfront list-distributions'.split())
         self.assertEqual(rc, 1)
-        self.assertIn(preview.PREVIEW_SERVICES['cloudfront'],
+        self.assertIn(preview.PreviewModeCommandMixin.HELP_SNIPPET,
                       self.stderr.getvalue())
 
-    @mock.patch('awscli.help.get_renderer')
-    def test_preview_service_off(self, get_renderer):
+    def test_preview_service_not_true(self):
+        # If it's not "true" then we still make it a preview service.
+        self.full_config['preview'] = {'cloudfront': 'false'}
+        rc = self.driver.main('cloudfront list-distributions'.split())
+        self.assertEqual(rc, 1)
+        self.assertIn(preview.PreviewModeCommandMixin.HELP_SNIPPET,
+                      self.stderr.getvalue())
+
+    def test_preview_service_enabled_makes_call(self):
         self.full_config['preview'] = {'cloudfront': 'true'}
+        # Registering this handler will prevent the actual service
+        # call to cloudfront.
+        self.driver.session.register('calling-command.cloudfront',
```

Comment:
Seems a little hacky to me (requires knowledge of the event system and these are more prone to changes). I would prefer to mock out the endpoint or use ``BaseAWSCommandParamsTest``.